sdk/PowerBI.Core.Tests/Security/PowerBITokenTests.cs

```diff
@@ -46,6 +49,41 @@ public void CanCreateReportEmbedToken()
         }
 
         [TestMethod]
+        public void CanCreateReportEmbedTokenForDatasetId()
+        {
+            var workspaceId = Guid.NewGuid().ToString();
+            var datasetId = Guid.NewGuid().ToString();
+
+            var token = PowerBIToken.CreateReportEmbedTokenWithDataset("Contoso", workspaceId, datasetId: datasetId, username: "TestUser", roles: new[] { "TestRole" });
+
+            Assert.IsNotNull(token);
+            var jwt = token.Generate(this.accessKey);
+            Assert.IsFalse(string.IsNullOrEmpty(jwt));
+
+            var decodedToken = new JwtSecurityToken(jwt);
+
+            var versionClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.Version);
+            var typeClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.JwtType);
+            var wcnClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.WorkspaceCollectionName);
+            var widClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.WorkspaceId);
+            var didCliam = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.DatasetId);
+            var usernameClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.Username);
+            var rolesClaim = decodedToken.Claims.FirstOrDefault(c => c.Type == PowerBIToken.ClaimTypes.Roles);
+
+            Assert.AreEqual("PowerBISDK", decodedToken.Issuer);
+            Assert.IsTrue(decodedToken.Audiences.Contains("https://analysis.windows.net/powerbi/api"));
+            Assert.IsTrue(decodedToken.ValidTo >= DateTime.UtcNow);
+            Assert.IsTrue(decodedToken.ValidTo <= DateTime.UtcNow.AddHours(1));
+            Assert.AreEqual("0.2.0", versionClaim.Value);
```

Comment:
Consider adding const variable for the version number and use it as the expected value throughout the different tests